src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright(c) 2020 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading.Tasks;
+using Tizen.NUI.BaseComponents;
+using Tizen.NUI.Binding;
+
+namespace Tizen.NUI.Components
+{
+    /// <summary>
+    /// The Navigator is a class which navigates pages with stack methods such
+    /// as Push and Pop.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class Navigator : Control
+    {
+        //This will be replaced with view transition class instance.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        protected Animation _curAnimation = null;
```

Comment:
@jaehyun0cho, I'm sorry for the late change requests.

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/field?redirectedfrom=MSDN
It is not recommended to have a protected field.
Why don't you make it to a `property` and have capitalized name? (e.g. CurrentAnimation)
And please add description in `<summary>` for protected APIs if possible.

And can I ask why these animations should be `protected`?

Reply:
Basically, the codes for transition in Navigator are temporary and should not be opened to public.
The transition in Navigator will be supported by ViewTransition which @hyunjushin is planning.
So I made animation handles protected and I did not write summary for that.
About using protected, I thought that the animation handles may be used in the NUI classes internally which inherit Navigator.

Since those are not supposed to be opened, I will make them private.
Thank you.